bemserver_ui/static/scripts/modules/views/timeseries/dataExplore.js

```diff
@@ -65,11 +65,77 @@ export class TimeseriesDataExploreView {
         this.#endDatetimePickerElmt.dateMin = this.#startDatetimePickerElmt.date;
     }
 
+    #addTsParamInputs(tsData) {
+        let tsParam = document.createElement("div");
+        tsParam.id = "tsParam-" + tsData.itemId;
+        
+        tsParam.innerHTML =
+        "<div class=\"row d-xl-flex d-grid m-2 mb-3\">"
+        + "<h6>"
+        + tsData.itemName
+        + "</h6>"
+        + "<div class=\"col pb-2 pb-xl-0\">"
+        + "<div class=\"input-group input-group-sm\">"
+        + "<span class=\"input-group-text\">Position</span>"
+        + "<select class=\"form-select\" id=\"position-"
+        + tsData.itemId
+        + "\" name=\"position\" aria-label=\"Select a position\">"
+        + "<option value=\"left\" selected>Left</option>"
+        + "<option value=\"right\">Right</option>"
+        + "</select>"
+        + "</div>"
+        + "</div>"
+        + "<div class=\"col pb-2 pb-xl-0\">"
+        + "<div class=\"input-group input-group-sm\">"
+        + "<span class=\"input-group-text\">Type</span>"
+        + "<select class=\"form-select\" id=\"type-"
+        + tsData.itemId
+        + "\" name=\"type\" aria-label=\"Select a type of graph\">"
+        + "<option value=\"line\" selected>Line chart</option>"
+        + "<option value=\"bar\">Bar chart</option>"
+        + "</select>"
+        + "</div>"
+        + "</div>"
+        + "<div class=\"col pb-2 pb-xl-0\">"
+        + "<div class=\"input-group input-group-sm\">"
+        + "<span class=\"input-group-text\">Style</span>"
+        + "<select class=\"form-select\" id=\"style-"
+        + tsData.itemId
+        + "\" name=\"style\" aria-label=\"Select a style\">"
+        + "<option value=\"solid\" selected>Solid</option>"
+        + "<option value=\"dashed\">Dashed</option>"
+        + "<option value=\"dotted\">Dotted</option>"
+        + "</select>"
+        + "</div>"
+        + "</div>"
+        + "<div class=\"col pb-2 pb-xl-0\">"
+        + "<div class=\"input-group input-group-sm\">"
+        + "<span class=\"input-group-text\">Color</span>"
+        + "<input type=\"color\" class=\"form-control form-control-color\" id=\"color-"
+        + tsData.itemId
+        + "\" name=\"color\" value=\""
+        + this.#chart.colors[(this.#tsSelector.selectedItemNames.length - 1) % (this.#chart.colors.length)]
+        + "\">"
+        + "</div>"
+        + "</div>";
+
+        document.getElementById("tsParam").appendChild(tsParam);
+    }
+
     #initEventListeners() {
         this.#tsSelector.addEventListener("toggleItem", (event) => {
             event.preventDefault();
 
             this.#updateLoadBtnState();
+
+            if (event.detail.itemIsActive) {
+                this.#addTsParamInputs(event.detail);
+            }
+            else {
+                delete this.#tsSelector[event.detail.itemName];
+
+                document.getElementById("tsParam-" + event.detail.itemId)?.remove();      
```

Comment:
According with suggestion above, here we could have this following code:
```
if (this.#tsParamInputElmts[event.detail.itemId] != null) {  // maybe this check is not useful
    this.#tsParamInputElmts[event.detail.itemId]["containerElmt"].remove();
    delete this.#tsParamInputElmts[event.detail.itemId];
}
```

Reply:
For this case and the other one, i made an if else condition because i want to add the component if the user as selected it and delete it if the user unselected it. And that's the way you told me to do it because i didn't remove the value in the list when i deleted the component. If i change it, the code doesn't work.